### PR TITLE
[scaling] deployer: add idle minutes logic

### PR DESCRIPTION
**Pull request by iulianbarbu**
## Description

This is rebased on top of #1038 .

* added idle minutes logic
* fixed a bug where the container was started but the health-check failed due to the container not being started completely

Todos:
* add an RPC to wake the service

## How has this been tested? (if applicable)

Manually locally.

### CODE REVIEW

File: deployer/src/project/task.rs [hunk 1]
```diff
@@ -141,15 +142,28 @@ pub fn start() -> impl Task<ServiceTaskContext, Output = ServiceState, Error = E
...
-                    .is_ok()
                 {
-                    TaskResult::Done(ServiceState::Ready(ready))
+                    if runtime_client.health_check(Ping {}).await.is_ok() {
+                        TaskResult::Done(ServiceState::Running(running))
+                    } else {
+                        TaskResult::Done(
+                            ServiceState::Running(running)
+                                .reboot()
+                                .expect("failed when tried to reboot the service"),
```
Comment (chesedo) [inline deployer/src/project/task.rs:158]:
```suggestion
                                .expect("failed when trying to reboot the service"),
```

File: deployer/src/project/task.rs [hunk 2]
```diff
@@ -141,15 +142,28 @@ pub fn start() -> impl Task<ServiceTaskContext, Output = ServiceState, Error = E
...
+                        )
+                    }
                 } else {
-                    TaskResult::Done(ServiceState::Ready(ready).reboot().unwrap())
+                    TaskResult::Done(
+                        ServiceState::Running(running)
+                            .reboot()
+                            .expect("failed when tried to reboot the service"),
```
Comment (chesedo) [inline deployer/src/project/task.rs:165]:
```suggestion
                            .expect("failed when trying to reboot the service"),
```

File: deployer/src/project/service/state/d_started.rs [hunk 3]
```diff
@@ -75,103 +57,34 @@ where
...
-                    .map_err(|_err| {
-                        ServiceErrored::internal("invalid `started_at` response from Docker daemon")
-                    })?;
-            let now = chrono::offset::Utc::now();
-            if started_at + chrono::Duration::seconds(120) < now {
-                return Err(ServiceErrored::internal(
-                    "project did not become healthy in time",
-                ));
+        {
+            // If healt-check works, move it to ready.
```
Comment (chesedo) [inline deployer/src/project/service/state/d_started.rs:65]:
```suggestion
            // If health-check works, move it to ready.
```

**Review by chesedo — APPROVED** [review]
LGTM :partying_face: 

nit: grammar suggestions

**Comment by iulianbarbu** [comment]
Closing this until clarifying better the approach on the new deployer.

**Pull request closed without merging**